### PR TITLE
merge streamplot_demo_features streamplot_demo_masking streamplot_demo_start_points

### PR DESCRIPTION
**Pull request by patniharshit**
refs #7956

### CODE REVIEW

File: examples/images_contours_and_fields/streamplot_demo_features.py [hunk 1]
```diff
@@ -31,3 +31,56 @@
 ax2.streamplot(X, Y, U, V, density=0.6, color='k', linewidth=lw)
 
 plt.show()
```
Comment (QuLogic) [inline examples/images_contours_and_fields/streamplot_demo_features.py]: Remove; there's one at the end.

File: examples/images_contours_and_fields/streamplot_demo_features.py [hunk 2]
```diff
@@ -31,3 +31,56 @@
...
+
+"""
+================================
+Streamplot function with masking
```
Comment (QuLogic) [inline examples/images_contours_and_fields/streamplot_demo_features.py]: You'll need to merge this docstring with the one at the beginning to make something coherent.

File: examples/images_contours_and_fields/streamplot_demo_features.py [hunk 3]
```diff
@@ -31,3 +31,56 @@
...
+================================
+
+This example shows how streamlines created by the streamplot function skips
+masked regions and NaN values.
+"""
+w = 3
+Y, X = np.mgrid[-w:w:100j, -w:w:100j]
```
Comment (QuLogic) [inline examples/images_contours_and_fields/streamplot_demo_features.py]: Can `X` and `Y` be reused from above?

File: examples/images_contours_and_fields/streamplot_demo_features.py [hunk 4]
```diff
@@ -31,3 +31,56 @@
...
+U[:20, :20] = np.nan
+U = np.ma.array(U, mask=mask)
+
+fig, ax = plt.subplots()
+ax.streamplot(X, Y, U, V, color='r')
+
+ax.imshow(~mask, extent=(-w, w, -w, w), alpha=0.5,
+          interpolation='nearest', cmap=plt.cm.gray)
+
+plt.show()
```
Comment (QuLogic) [inline examples/images_contours_and_fields/streamplot_demo_features.py]: Remove.

File: examples/images_contours_and_fields/streamplot_demo_features.py [hunk 5]
```diff
@@ -31,3 +31,56 @@
...
+
+"""
+========================================
+Streamplot function with starting points
```
Comment (QuLogic) [inline examples/images_contours_and_fields/streamplot_demo_features.py]: Again, you need to merge this with the first docstring.

File: examples/images_contours_and_fields/streamplot_demo_features.py [hunk 6]
```diff
@@ -31,3 +31,56 @@
...
+========================================
+
+This example shows how to fix the streamlines that are plotted, by passing
+an array of seed points to the `start_points` keyword argument.
+"""
+Y, X = np.mgrid[-3:3:100j, -3:3:100j]
+U = -1 - X**2 + Y
+V = 1 + X - Y**2
```
Comment (QuLogic) [inline examples/images_contours_and_fields/streamplot_demo_features.py]: Can any of these be reused from before?

**Comment by afvincent** [comment]
Wouldn't this be a duplicate effort of #8082?

**Comment by patniharshit** [comment]
@afvincent  Should I continue with the suggested review request?

**Comment by tacaswell** [comment]
I suggest combining efforts with @tmcclintock  .  The comments between the two PRs are very similar.  I suggest starting from the branch in  #8082 , addressing both sets of comments and opening a new PR with the combined efforts.

**Comment by afvincent** [comment]
@patniharshit I agree with @tacaswell on the idea of merging the two PRs. In the case where some review comments about English are overlapping between @QuLogic's ones and mine, just trust @QuLogic ;).

**Comment by QuLogic** [comment]
Replaced by #8336.

**Pull request closed without merging**